crates/evm/core/src/backend/mod.rs

```diff
@@ -1350,31 +1355,26 @@ impl DatabaseExt for Backend {
         )
     }
 
+    // Note(zk): This function in upstream code is not implemented as part of the strategy pattern,
+    // but is instead a standalone function. We have moved it here to make it part of the
+    // strategy, as there is some abstraction in the middle since the envelopes, and types are
+    // different. The changes are:
+    // - The function signature has been changed to take a `Bytes` instead of `the
+    //   TransactionRequest`
+    // - The function signature has been changed to return a `TransactionMaybeSigned` instead of
+    //   empty tuple
+    // - Avoids some cloning of the backend state
+    // See new transact_from_tx evm implementation in: crates/evm/core/src/backend/strategy.rs
+    // See zk implementation in: crates/strategy/zksync/src/backend/runner.rs
```

Comment:
Love the detailing the changes here! I think we could apply it in more places to make it _extra_ clear! Awesome idea.
```suggestion
    // NOTE(zk): We implement this function via strategy pattern, letting the strategy decide how to decode the data.
    // The changes are:
    // - `tx` type changed from`TransactionRequest` to `Bytes`, and renamed to `data`
    // - function now returns `TransactionMaybeSigned`
```